beetsplug/export.py

```diff
@@ -0,0 +1,145 @@
+# -*- coding: utf-8 -*-
+# This file is part of beets.
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+
+"""Exports data from beets
+"""
+
+from __future__ import division, absolute_import, print_function
+
+import json
+import codecs
+
+from datetime import datetime, date
+from beets.plugins import BeetsPlugin
+from beets import ui
+from beets import mediafile
+from beetsplug.info import make_key_filter, library_data, tag_data
+
+json.JSONEncoder.default = lambda self, obj: \
+    (obj.isoformat() if isinstance(obj, datetime) or isinstance(obj, date)
+     else None)
+
+
+class ExportPlugin(BeetsPlugin):
+
+    def __init__(self):
+        super(ExportPlugin, self).__init__()
+
+        self.config.add({
+            'default_format': 'json',
+            'json': {
+                # json module formatting options
+                'formatting': {
+                    'ensure_ascii': False,
+                    'indent': 4,
+                    'separators': (',', ': '),
+                }
+            },
+            # TODO: Use something like the edit plugin
+            # 'item_fields': []
+        })
+
+    def commands(self):
+        # TODO: Add option to use albums
+
+        cmd = ui.Subcommand('export', help=u'export data from beets')
+        cmd.func = self.run
+        cmd.parser.add_option(
+            u'-l', u'--library', action='store_true',
+            help=u'show library fields instead of tags',
+        )
+        cmd.parser.add_option(
+            u'--append', action='store_true', default=False,
+            help=u'if should append data to the file',
+        )
+        cmd.parser.add_option(
+            u'-i', u'--include-keys', default=[],
+            action='append', dest='included_keys',
+            help=u'comma separated list of keys to show',
+        )
+        cmd.parser.add_option(
+            u'-p', u'--path',
+            help=u'path for the output file. If not given, will print the data'
+        )
```

Comment:
I used this because I use Windows (mostly) and just don't want to deal with the possible encoding problems.
Should `-o` have a long option like `--out/--output`?